rpc/internal/srv/srv.go

```diff
@@ -17,6 +19,13 @@ type SearchArgs struct {
 	Opts *zoekt.SearchOptions
 }
 
+// We need a different type for JSON requests because the query needs to be
+// parsed.
+type JSONSearchArgs struct {
+	Q    string
+	Opts *zoekt.SearchOptions
+}
```

Comment:
`rpc` is not necessarily an appropriate package for this to live in.  Suggestions welcome.